src/probnum/_config.py

```diff
@@ -36,13 +42,38 @@ class Configuration:
     0.01
     """
 
+    def __init__(self) -> None:
+        object.__setattr__(self, "_options_registry", dict())
```

Comment:
We have to use the "default" attribute write access here, since we override `__setattr__`.

Reply:
And for a reason I haven't quite figured out, yet, using `self.__dict__`, instead of a dictionary that is initialized as an additional class attribute (i.e., here, `self._options_registry`), does not work.
If somebody knows why, I would be very grateful to learn.